From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: bot-for-go[bot] <199222863+bot-for-go[bot]@users.noreply.github.com>
Date: Fri, 3 Jun 2022 12:26:05 -0500
Subject: [PATCH] Use MCR mirror for buildpack-deps

---
 Dockerfile-linux.template | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Dockerfile-linux.template b/Dockerfile-linux.template
index e14bb6a6bf3344..0e18cc03ae5a63 100644
--- a/Dockerfile-linux.template
+++ b/Dockerfile-linux.template
@@ -39,7 +39,7 @@ RUN set -eux; \
 	; \
 	tdnf clean all
 {{ ) else ( -}}
-FROM buildpack-deps:{{ env.variant }}-scm AS build
+FROM mcr.microsoft.com/mirror/docker/library/buildpack-deps:{{ env.variant }}-scm AS build
 {{ ) end -}}
 
 ENV PATH /usr/local/go/bin:$PATH
@@ -310,7 +310,7 @@ RUN set -eux; \
 	tdnf clean all
 
 {{ ) else ( -}}
-FROM buildpack-deps:{{ env.variant }}-scm
+FROM mcr.microsoft.com/mirror/docker/library/buildpack-deps:{{ env.variant }}-scm
 
 # install cgo-related dependencies
 RUN set -eux; \
