From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: microsoft-golang-bot <microsoft-golang-bot@users.noreply.github.com>
Date: Fri, 3 Jun 2022 12:26:05 -0500
Subject: [PATCH] Use MCR mirror for buildpack-deps

---
 Dockerfile-linux.template | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Dockerfile-linux.template b/Dockerfile-linux.template
index b8849cc2142d76..d71d53dde7565e 100644
--- a/Dockerfile-linux.template
+++ b/Dockerfile-linux.template
@@ -30,7 +30,7 @@ RUN set -eux; \
 	; \
 	tdnf clean all
 {{ ) else ( -}}
-FROM buildpack-deps:{{ env.variant }}-scm AS build
+FROM mcr.microsoft.com/mirror/docker/library/buildpack-deps:{{ env.variant }}-scm AS build
 {{ ) end -}}
 
 ENV PATH /usr/local/go/bin:$PATH
@@ -209,7 +209,7 @@ RUN set -eux; \
 	tdnf clean all
 
 {{ ) else ( -}}
-FROM buildpack-deps:{{ env.variant }}-scm
+FROM mcr.microsoft.com/mirror/docker/library/buildpack-deps:{{ env.variant }}-scm
 
 # install cgo-related dependencies
 RUN set -eux; \
