From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: microsoft-golang-bot <microsoft-golang-bot@users.noreply.github.com>
Date: Fri, 3 Jun 2022 12:26:05 -0500
Subject: [PATCH] Use MCR mirror for buildpack-deps

---
 Dockerfile-linux.template | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Dockerfile-linux.template b/Dockerfile-linux.template
index 58392a2e0d3205..0e95c066e38cec 100644
--- a/Dockerfile-linux.template
+++ b/Dockerfile-linux.template
@@ -20,7 +20,7 @@ FROM alpine:{{ alpine_version }} AS build
 {{ ) elif is_cbl_mariner then ( -}}
 FROM mcr.microsoft.com/cbl-mariner/base/core:{{ cbl_mariner_version }}
 {{ ) else ( -}}
-FROM buildpack-deps:{{ env.variant }}-scm AS build
+FROM mcr.microsoft.com/mirror/docker/library/buildpack-deps:{{ env.variant }}-scm AS build
 {{ ) end -}}
 
 ENV PATH /usr/local/go/bin:$PATH
@@ -199,7 +199,7 @@ RUN set -eux; \
 	tdnf clean all
 
 {{ ) else ( -}}
-FROM buildpack-deps:{{ env.variant }}-scm
+FROM mcr.microsoft.com/mirror/docker/library/buildpack-deps:{{ env.variant }}-scm
 
 # install cgo-related dependencies
 RUN set -eux; \
