From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: bot-for-go[bot] <199222863+bot-for-go[bot]@users.noreply.github.com>
Date: Mon, 21 Jul 2025 08:39:31 -0700
Subject: [PATCH] Remove packages marked vulnerable

---
 Dockerfile-linux.template | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/Dockerfile-linux.template b/Dockerfile-linux.template
index bfa9b58f6558b3..b380f4f4041422 100644
--- a/Dockerfile-linux.template
+++ b/Dockerfile-linux.template
@@ -311,6 +311,16 @@ RUN set -eux; \
 {{ ) else ( -}}
 FROM mcr.microsoft.com/mirror/docker/library/buildpack-deps:{{ env.variant }}-scm
 
+# Remove optional packages that are marked as vulnerable and that don't have fixes available.
+# Then try to install up-to-date versions, so that if the distro makes them available in the future,
+# our next build will automatically include them.
+RUN set -eux; \
+	apt-get autoremove -y \
+		subversion \
+	; \
+	apt-get satisfy -y 'subversion (>= 1.14.5)' || echo 'patched subversion not available' ; \
+	rm -rf /var/lib/apt/lists/*
+
 # install cgo-related dependencies
 RUN set -eux; \
 	apt-get update; \
